### PR TITLE
Migrate PathDisplay

### LINKED ISSUE #81: Migrate PathDisplay

### PR DESCRIPTION
**Pull request by greimela-si**
Resolves #81.

* Migrate PathDisplay to be used in RViz2.
* Refactor PathDisplay
* Fix alpha blending
* Add tests

### CODE REVIEW

**Comment by greimela-si** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/4146/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/1223/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/3427/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/4250/)

File: rviz_default_plugins/CMakeLists.txt [hunk 1]
```diff
@@ -118,6 +121,7 @@ target_link_libraries(rviz_default_plugins
 )
 ament_target_dependencies(rviz_default_plugins
   # geometry_msgs
```
Comment (wjwwood) [inline rviz_default_plugins/CMakeLists.txt]: This one should probably also be uncommented?

File: rviz_default_plugins/package.xml [hunk 2]
```diff
@@ -29,6 +29,8 @@
   <depend>rviz_common</depend>
   <depend>rviz_rendering</depend>
   <depend>urdf</depend>
+  <depend>geometry_msgs</depend>
+  <depend>nav_msgs</depend>
```
Comment (wjwwood) [inline rviz_default_plugins/package.xml]: nitpick: please alphabetically order dependencies :)

File: rviz_default_plugins/src/rviz_default_plugins/displays/path/path_display.hpp [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+
```
Comment (wjwwood) [inline rviz_default_plugins/src/rviz_default_plugins/displays/path/path_display.hpp]: nitpick: minimize vertical whitespace

File: rviz_default_plugins/test/rviz_default_plugins/displays/path/path_display_test.cpp [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+
```
Comment (wjwwood) [inline rviz_default_plugins/test/rviz_default_plugins/displays/path/path_display_test.cpp]: nitpick: mvw

**Review by wjwwood — APPROVED** [review]
lgtm, with some minor comments

**Pull request merged**